File: scala-implicits/src/main/scala/com/couchbase/client/scala/kv/MutateInMacro.scala
package com.couchbase.client.scala.kv

sealed trait MutateInMacro {
  private[scala] def value: String
}

object MutateInMacro {

  case object CAS extends MutateInMacro {
    private[scala] val value = """"${Mutation.CAS}""""
  }

  case object SeqNo extends MutateInMacro {
    private[scala] val value = """"${Mutation.seqno}""""
  }

  case object ValueCrc32c extends MutateInMacro {
    private[scala] val value = """"${Mutation.value_crc32c}""""
  }
}

File: kotlin-client/src/main/kotlin/com/couchbase/client/kotlin/kv/MutateInMacro.kt
package com.couchbase.client.kotlin.kv

import com.couchbase.client.kotlin.annotations.VolatileCouchbaseApi

@VolatileCouchbaseApi
public open class MutateInMacro internal constructor(internal val value: String) {
    public object Cas : MutateInMacro("\${Mutation.CAS}")
    public object SeqNo : MutateInMacro("\${Mutation.seqno}")
    public object ValueCrc32c : MutateInMacro("\${Mutation.value_crc32c}")

    override fun toString(): String {
        return "MutateInMacro(value='$value')"
    }
}
